enhancements/update/update-status-api.md

```diff
@@ -0,0 +1,410 @@
+  # one node per pool
+  - name: mcp-1
+    informers:
+    # one node per informer reporting pool insights, typically ~2-5
+    - name: informer-2
+      insights
+      # one leaf per insight
+```
+
+Health insights are not expected to be created in large numbers (like alerts, they are not helpful when too many are reported). A healthy update should have zero. The aspiration for the system is to produce precisely one health insight for a problematic update: the one that clearly describes the problem affecting the update. We are unlikely to achieve this ideal, but a typical problematic update should produce no more than ~10 health insights.
+
+However, there is a single Status Insight for each resource involved in the update, so we can expect `1 + # of ClusterOperators + # of MachineConfigPools + # of Nodes` status insights. That means that for big clusters, there will always be (lower) hundreds of insights, primarily because of Nodes.
+
+The Status Insights are not expected to be large data structures, but some contain Kubernetes conditions, so they cannot be considered entirely trivial, and there can be a high number of them. The Health Insights are larger because they can carry a longer-form description of the issue, expected to be around a paragraph of text.
+
+For an API this large, we need to consider the possibility of hitting Kubernetes API resource limits. Even with too many long insights, hitting the 1.5MiB limit does not seem plausible under regular operation. We can protect against the risk by setting appropriate bounds and limits on API fields (especially lists and the long-form description field of health insights), and these limits will be part of the contract for insight producers.
+
+The size and complexity also impact the readability of the API for humans. This is further discussed in [Drawbacks][this-drawbacks].
+
+### Drawbacks
+
+The pattern of the CVO directly deploying a non-operator component is unusual in OpenShift. We could introduce an entirely new Cluster Operator to manage the USC, but because the update functionality is so closely tied to the CVO, an additional layer seems excessive and unnecessary. Adding this layer can be considered in the future if the proposed model is problematic.
+
+Placing the cluster inspection logic directly into the USC puts the OTA team in a position where we need to maintain the logic while not being experts in a significant part of the domain (interpreting `MachineConfigPool` and `Node` states). This is fine as long as we eventually move the logic to the Machine Config Operator to serve as an insight producer, which depends on the future architecture of the system.
+
+The API-backed `oc adm upgrade status` will lose the ability to run against older clusters that do not have the API (or against non-TechPreview clusters while the feature is still in TechPreview). The feature was requested to be an API from the start, and the client-based prototype was meant to be a temporary solution.
```

Comment:
Do you have a link to the original request? Might be useful to inline a link here to it

Reply:
I do not have a single distilled resource I could include here. There's a trail of RFE/OCPSTRAT JIRAs:

https://issues.redhat.com/browse/RFE-4839
https://issues.redhat.com/browse/OCPSTRAT-1823 

I'll include these links in the enhancement.